frontend/src/components/AccountDetailsEditContainer.vue

```diff
@@ -0,0 +1,225 @@
+              Account gespeichert
+            </div>
+          </div>
+          <ButtonComponent
+            class="w-fit p-1"
+            :loading="disableInputShowSpinner"
+            @click="updateInformation()"
+          >
+            Speichern
+          </ButtonComponent>
+        </div>
+        <div :class="showadminview ? 'flex items-center pt-1' : 'collapse'">
+          <div class="flex-grow text-center">
+            <div class="text-green-500 italic" v-if="deleted">
+              Account gelöscht
+            </div>
+          </div>
+          <ButtonComponent
+            class="w-fit p-1"
+            :loading="disableInputShowSpinner"
+            @click="deleteAccount()"
+          >
+            Löschen
+          </ButtonComponent>
+        </div>
+      </template>
+    </ContainerComponent>
+  </div>
+</template>
+<script setup lang="ts">
+import { ref } from 'vue';
+import TextInputComponent from './TextInputComponent.vue';
+import ButtonComponent from './ButtonComponent.vue';
+import ContainerComponent from './ContainerComponent.vue';
+import { IUser, UserClass } from '../../../model/user/IUser';
+import { AccountRestInterface } from '../restInterfaces/AccountRestInterface';
+
+const props = defineProps({
+  /**
+   * Whether the delete button is displayed
+   */
+  showadminview: {
+    type: Boolean,
+    required: false,
+    default: false,
+  },
+  /**
+   * The user to display
+   */
+  user: {
+    type: Object as () => Partial<IUser>,
+    required: true,
+  },
+});
+
+const nameField = ref(props.user.name);
+const mailField = ref(props.user.mail);
+const pwField = ref('');
+const pwFieldControl = ref('');
+const disableInputShowSpinner = ref(false);
+const deleted = ref(false);
+const displayError = ref(false);
+const displaySuccess = ref(false);
+let originalUser = props.user as IUser;
+const isAdmin = ref(originalUser.type === UserClass.ADMIN);
+/**
+ * toggles the checkbox
+ */
+function toggleCheckBox(): void {
+  if (!disableInputShowSpinner.value) {
+    isAdmin.value = !isAdmin.value;
+  }
+}
+
+/**
+ * Updates Information from the User with the entered information
+ */
+async function updateInformation(): Promise<void> {
+  if (deleted.value) {
+    return;
+  }
+  displaySuccess.value = false;
+  displayError.value = false;
+  disableInputShowSpinner.value = true;
+
+  const updateUser: Partial<IUser> = { id: originalUser.id };
+
+  if (nameField.value !== '') {
+    updateUser.name = nameField.value;
+  }
+  if (mailField.value !== '') {
+    updateUser.mail = mailField.value;
+  }
+  if (pwField.value !== '') {
+    if (pwField.value === pwFieldControl.value) {
+      updateUser.password = pwField.value;
+    } else {
+      displayError.value = true;
+      disableInputShowSpinner.value = false;
+      return;
+    }
+  }
+  if (props.showadminview) {
+    updateUser.type = isAdmin.value ? UserClass.ADMIN : UserClass.REGULAR;
+  }
+  await AccountRestInterface.updateAccount(updateUser);
```

Comment:
I still think the rest interfaces have no place here.
Vue Documentation: 
> Components allow us to split the UI into independent and reusable pieces, and think about each piece in isolation.

This would not be the case, since it is heavily tied to this function of the rest interfaces

Reply:
since the button should always do the same in all components, namely save the user, they should not have the option to get rebind i.e. to a deleting call.
also taking logic out of the component would lead to copy paste, what would be much worse than violating a dokumentation(which also more sounds like a hint or recommendation) , because it could lead to massive inconsistencies

Reply:
I do agree with the point that e.g. a save button won't logically be bound to anything other than our save function, in my opinion, the container cant logically work on its own without the save/delete functionality

Also, if we need to change the signature of our save function, we'd only have to change it in this location, instead of rechecking every location for the save function anywhere, and fixing it

Reply:
Status update: We decided to allow restinterface calls in compnents. Resolving this conversation